### ISSUE TITLE
Add ability to scroll or Make tty Height totally flexible ( Feature Request )

### ISSUE THREAD

**Issue by xd003** [Adikso/BashBot#47]
I am running a bash script through BashBot and the script has quite many options to be displayed . BashBot generated tty in discord only lets me see the bottom half of the script ( it first shows the top half and then immediately shows the bottom half )

So is it possible to add a scrollbar to the terminal .If not , you can make the tty screen height unlimited ( ie. When a user opens tty and starts any process , it will display everything from start to end as be as long as it needs to be ).  I don't know if you might have a better solution to this problem but it would really great have this issue addressed

Thank you

**Comment by xd003** [comment]
Hello @Adikso  , can you please have a look at this issue

**Comment by Adikso** [comment]
I can't make unlimited terminal output because library that I am using for rendering terminal (which is needed to support interactive applications) does not support it. I can't add scrollbar because it is not possible on discord but I can add scroll up and down commands which you could later assign via control command

**Comment by xd003** [comment]
That's good to know that atleast something can be done. Scroll up and down buttons would definitely help . Looking forward to it

**Comment by xd003** [comment]
Hello sir @Adikso 
Any Update on this feature ?

**Comment by yuan-miranda** [comment]
> I can't add scrollbar because it is not possible on discord but I can add scroll up and down commands which you could later assign via control command

@Adikso thats a great idea, another one is, in my basic implementation of Shell bot like you did, I made the bot send a file output as the command output logger, did it by taking the stdout stream (or the `stream` from conn,shell() in ssh2 js) and appending it on the file after 1-5 seconds of command execution. You could add some sort of interactive button or command for it for the bot to send.

tldr: I have no idea how complex your bot is when sending outputs, I didnt tried it yet

**Comment by Adikso** [comment]
> > I can't add scrollbar because it is not possible on discord but I can add scroll up and down commands which you could later assign via control command
> 
> @Adikso thats a great idea, another one is, in my basic implementation of Shell bot like you did, I made the bot send a file output as the command output logger, did it by taking the stdout stream (or the `stream` from conn,shell() in ssh2 js) and appending it on the file after 1-5 seconds of command execution. You could add some sort of interactive button or command for it for the bot to send.
> 
> tldr: I have no idea how complex your bot is when sending outputs, I didnt tried it yet

I don't exactly understand what you mean. This bot is sending you output realtime the way it would happen in normal terminal app

**Comment by yuan-miranda** [comment]
> > > I can't add scrollbar because it is not possible on discord but I can add scroll up and down commands which you could later assign via control command
> > 
> > 
> > @Adikso thats a great idea, another one is, in my basic implementation of Shell bot like you did, I made the bot send a file output as the command output logger, did it by taking the stdout stream (or the `stream` from conn,shell() in ssh2 js) and appending it on the file after 1-5 seconds of command execution. You could add some sort of interactive button or command for it for the bot to send.
> > tldr: I have no idea how complex your bot is when sending outputs, I didnt tried it yet
> 
> I don't exactly understand what you mean. This bot is sending you output realtime the way it would happen in normal terminal app

ahh I thought its not updating the output automatically, anyway is the source code still working? I wanna try it out!

**Comment by Adikso** [comment]
> > > > I can't add scrollbar because it is not possible on discord but I can add scroll up and down commands which you could later assign via control command
> > > 
> > > 
> > > @Adikso thats a great idea, another one is, in my basic implementation of Shell bot like you did, I made the bot send a file output as the command output logger, did it by taking the stdout stream (or the `stream` from conn,shell() in ssh2 js) and appending it on the file after 1-5 seconds of command execution. You could add some sort of interactive button or command for it for the bot to send.
> > > tldr: I have no idea how complex your bot is when sending outputs, I didnt tried it yet
> > 
> > 
> > I don't exactly understand what you mean. This bot is sending you output realtime the way it would happen in normal terminal app
> 
> ahh I thought its not updating the output automatically, anyway is the source code still working? I wanna try it out!

It worked the last time that I checked
